sphinx/ext/autodoc/__init__.py

```diff
@@ -1710,6 +1726,14 @@ def format_signature(self, **kwargs: Any) -> str:
                     documenter.object = func
                     documenter.objpath = [None]
                     sigs.append(documenter.format_signature())
+        if overloaded:
+            for overload in self.analyzer.overloads.get('.'.join(self.objpath)):
+                if not inspect.isstaticmethod(self.object, cls=self.parent,
+                                              name=self.object_name):
+                    parameters = list(overload.parameters.values())
+                    overload = overload.replace(parameters=parameters[1:])
+                sig = stringify_signature(overload, **kwargs)
+                sigs.append(sig)
```

Comment:
Could eliminate the `if` if you wanted to.
```suggestion
        for overload in self.analyzer.overloads.get('.'.join(self.objpath), []):
            if not inspect.isstaticmethod(self.object, cls=self.parent,
                                          name=self.object_name):
                parameters = list(overload.parameters.values())
                overload = overload.replace(parameters=parameters[1:])
            sig = stringify_signature(overload, **kwargs)
            sigs.append(sig)
```

Reply:
No, it would be crashed with C modules. Sometimes `self.analyzer` becomes None. The `overloaded` is used to skip the check.